src/disc/omegah/Albany_OmegahMeshFieldAccessor.cpp

```diff
@@ -362,4 +361,22 @@ saveVector (const Thyra_Vector&  field_vector,
   m_mesh->set_tag(dim,field_name,read(mesh_data_h.write()),false);
 }
 
+void OmegahMeshFieldAccessor::reset_mesh_tags ()
+{
+  for (auto& [name, tag_handle] : m_tags) {
+    auto& array = tag_handle.array;
+    int dim     = tag_handle.ent_dim;
+    int ncmp    = tag_handle.ncomps;
+    if (m_mesh->has_tag(dim,name)) {
+      auto tag = m_mesh->get_tag<ST>(dim,name);
+      array = Omega_h::Write<ST>(tag->array().size(),name);
+      Kokkos::deep_copy(array.view(),tag->array().view());
```

Comment:
I think we could simplify this a bit with:

```
array = Omega_h::Write<ST>(tag->array().view());
```

Note, this and the following `set_tag` call should just increase the reference count on the original array/view (i.e., no copies).